integration/tsconfig-test.ts

```diff
@@ -0,0 +1,125 @@
+  return JSON5.parse(config);
+}
+
+const DEFAULT_CONFIG = {
+  include: ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
+  compilerOptions: {
+    allowJs: true,
+    esModuleInterop: true,
+    forceConsistentCasingInFileNames: true,
+    isolatedModules: true,
+    jsx: "react-jsx",
+    lib: ["DOM", "DOM.Iterable", "ES2019"],
+    moduleResolution: "node",
+    noEmit: true,
+    resolveJsonModule: true,
+    strict: true,
+    target: "es2019",
+    baseUrl: ".",
+    paths: {
+      "~/*": ["./app/*"],
+    },
+  },
+};
+
+// besides baseUrl, due to an upstream issue in tsconfig-paths
+// https://github.com/dividab/tsconfig-paths/pull/208
+test("should output default tsconfig if file is empty", async () => {
+  let fixture = await createFixture({
+    files: {
+      "tsconfig.json": json({ compilerOptions: { baseUrl: "." } }),
+    },
+  });
+
+  let tsconfig = await getTsConfig(fixture.projectDir);
+  expect(tsconfig).toEqual(DEFAULT_CONFIG);
+});
+
+test("should add/update mandatory config", async () => {
+  let fixture = await createFixture({
+    files: {
+      "tsconfig.json": json({
+        ...DEFAULT_CONFIG,
+        compilerOptions: {
+          ...DEFAULT_CONFIG.compilerOptions,
+          isolatedModules: false, // true is required by esbuild
+          // moduleResolution: "node", // this is required by esbuild
+        },
+      }),
+    },
+  });
+
+  let tsconfig = await getTsConfig(fixture.projectDir);
+  expect(tsconfig).toEqual(DEFAULT_CONFIG);
+});
+
+test("shouldn't change suggested config if set", async () => {
+  let config = {
+    ...DEFAULT_CONFIG,
+    compilerOptions: {
+      ...DEFAULT_CONFIG.compilerOptions,
+      strict: false,
+    },
+  };
+
+  let fixture = await createFixture({
+    files: {
+      "tsconfig.json": json(config),
+    },
+  });
+
+  let tsconfig = await getTsConfig(fixture.projectDir);
+  expect(tsconfig).toEqual(config);
+});
+
+test("allows for `extends` in tsconfig", async () => {
+  let config = {
+    extends: "./tsconfig.base.json",
+  };
+
+  let baseConfig = {
+    compilerOptions: {
+      allowJs: true,
+      baseUrl: ".",
+    },
+  };
+
+  let fixture = await createFixture({
+    files: {
+      "tsconfig.json": json(config),
+      "tsconfig.base.json": json(baseConfig),
+    },
+  });
+
+  let tsconfig = await getTsConfig(fixture.projectDir);
+  // our base config only sets a few options, so our local config should fill in the missing ones
+  expect(tsconfig).toEqual({
+    extends: "./tsconfig.base.json",
+    include: ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
+    compilerOptions: {
+      esModuleInterop: true,
+      forceConsistentCasingInFileNames: true,
+      isolatedModules: true,
+      jsx: "react-jsx",
+      lib: ["DOM", "DOM.Iterable", "ES2019"],
+      moduleResolution: "node",
+      noEmit: true,
+      resolveJsonModule: true,
+      strict: true,
+      target: "es2019",
+      paths: {
+        "~/*": ["./app/*"],
+      },
+    },
```

Comment:
Where do all these values come from? Is there a way to indicate where they come from by the test code itself?